tools/copyright.go

```diff
@@ -0,0 +1,66 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"sync"
+)
+
+// getCopyrightYear get copyright year from git log and file modify time.
+func getCopyrightYear(filePath string) (string, error) {
+	file, err := os.Stat(filePath)
+	if err != nil {
+		return "", err
+	}
+	modYear := fmt.Sprintf("%d", file.ModTime().Year())
+	cmd := exec.Command("bash", "-c", "git log --follow --format=%cd --date=format:%Y "+filePath+" | sort -u")
+	output, err := cmd.Output()
+	if err != nil {
+		return "", err
+	}
+	lines := strings.Split(strings.TrimSpace(string(output)), "\n")
+	if lines[0] != "" && lines[0] != modYear {
+		return lines[0] + "-" + modYear, nil
+	}
+	return modYear, nil
+}
+
+// main function to traveser docs folder and update copyright year.
+func main() {
+	var wg sync.WaitGroup
+	err := filepath.Walk("docs/", func(path string, info os.FileInfo, err error) error {
+		if err != nil || info.IsDir() {
+			return err
+		}
+		wg.Add(1)
+		go func() {
```

Comment:
Is there any reason for parallelizing this operation and hence ignoring all the errors within? If the time taken isn't significant, won't it be better to get all error messages?

Reply:
we may return the error